ipaclient/plugins/csrgen.py

```diff
@@ -75,10 +75,19 @@ def execute(self, *args, **options):
             util.check_writable_file(options['out'])
 
         principal = options.get('principal')
+        if principal is None:
+            raise errors.InvocationError(message=u"Principal is required")
         profile_id = options.get('profile_id')
         if profile_id is None:
             profile_id = dogtag.DEFAULT_PROFILE
         helper = options.get('helper')
+        if helper is None:
+            raise errors.InvocationError(message=u"CSR generation tool is"
```

Comment:
You are missing a trailing space in this line. You can make the code more readable by indenting the class' arguments:

```
            raise errors.InvocationError(
                message=_(u"CSR generation tool is required")
            )
```